// react-bulma-components/lib/components/modal/components/card/head.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _button = require('../../../button');

var _button2 = _interopRequireDefault(_button);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var ModalCardHead = function (_PureComponent) {
  _inherits(ModalCardHead, _PureComponent);

  function ModalCardHead() {
    _classCallCheck(this, ModalCardHead);

    return _possibleConstructorReturn(this, (ModalCardHead.__proto__ || Object.getPrototypeOf(ModalCardHead)).apply(this, arguments));
  }

  _createClass(ModalCardHead, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          children = _props.children,
          className = _props.className,
          showClose = _props.showClose,
          onClose = _props.onClose,
          props = _objectWithoutProperties(_props, ['children', 'className', 'showClose', 'onClose']);

      return _react2.default.createElement(
        'header',
        _extends({}, props, {
          className: (0, _classnames2.default)('modal-card-head', className)
        }),
        children,
        showClose && _react2.default.createElement(_button2.default, { remove: true, onClick: onClose })
      );
    }
  }]);

  return ModalCardHead;
}(_react.PureComponent);

ModalCardHead.displayName = 'Modal.Card.Head';
ModalCardHead.propTypes = {
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  style: _propTypes2.default.shape({}),
  showClose: _propTypes2.default.bool,
  onClose: _propTypes2.default.func
};
ModalCardHead.defaultProps = {
  children: null,
  className: '',
  style: {},
  showClose: true,
  onClose: null
};
exports.default = ModalCardHead;
