// react-bulma-components/lib/components/card/card.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _image = require('./components/image');

var _image2 = _interopRequireDefault(_image);

var _content = require('./components/content');

var _content2 = _interopRequireDefault(_content);

var _header = require('./components/header');

var _header2 = _interopRequireDefault(_header);

var _footer = require('./components/footer');

var _footer2 = _interopRequireDefault(_footer);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Card = function (_PureComponent) {
  _inherits(Card, _PureComponent);

  function Card() {
    _classCallCheck(this, Card);

    return _possibleConstructorReturn(this, (Card.__proto__ || Object.getPrototypeOf(Card)).apply(this, arguments));
  }

  _createClass(Card, [{
    key: 'render',
    value: function render() {
      var _classnames;

      var _props = this.props,
          className = _props.className,
          children = _props.children,
          type = _props.type,
          renderAs = _props.renderAs,
          props = _objectWithoutProperties(_props, ['className', 'children', 'type', 'renderAs']);

      if (type) {
        // eslint-disable-next-line no-console
        console.warn('Deprecation Warning: Prop type is deprecated and will be removed on future release, please use Card.Header, Card.Content and Card.Footer elements instead');
      }
      if (type === 'image') {
        return _react2.default.createElement(_image2.default, _extends({}, props, { className: className }));
      }
      var Element = renderAs;
      return _react2.default.createElement(
        Element,
        _extends({
          className: (0, _classnames3.default)(className, (_classnames = {}, _defineProperty(_classnames, 'card-' + type, type), _defineProperty(_classnames, 'card', !type), _classnames))
        }, props),
        children
      );
    }
  }]);

  return Card;
}(_react.PureComponent);

Card.propTypes = {
  className: _propTypes2.default.string,
  children: _propTypes2.default.node,
  style: _propTypes2.default.shape({}),
  renderAs: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.func]),
  /**
   * @deprecated please use Card.Header, Card.Content, Card.Footer instead
   */
  type: _propTypes2.default.oneOf(['header', 'header-item', 'header-icon', 'footer', 'footer-item', 'image', 'content'])
};
Card.defaultProps = {
  className: '',
  children: null,
  style: {},
  type: null,
  renderAs: 'div'
};
Card.Image = _image2.default;
Card.Content = _content2.default;
Card.Header = _header2.default;
Card.Footer = _footer2.default;
exports.default = Card;
