// react-bulma-components/lib/components/menu/components/list/components/item.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _list = require('../list');

var _list2 = _interopRequireDefault(_list);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var MenuListItem = function (_PureComponent) {
  _inherits(MenuListItem, _PureComponent);

  function MenuListItem() {
    _classCallCheck(this, MenuListItem);

    return _possibleConstructorReturn(this, (MenuListItem.__proto__ || Object.getPrototypeOf(MenuListItem)).apply(this, arguments));
  }

  _createClass(MenuListItem, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          children = _props.children,
          active = _props.active,
          className = _props.className,
          renderAs = _props.renderAs,
          props = _objectWithoutProperties(_props, ['children', 'active', 'className', 'renderAs']);

      var Element = renderAs;
      if (typeof children === 'string') {
        return _react2.default.createElement(
          'li',
          null,
          _react2.default.createElement(
            Element,
            _extends({ className: (0, _classnames2.default)(className, { 'is-active': active }) }, props),
            children
          )
        );
      }

      if (_react2.default.Children.only(children).type === _list2.default) {
        var child = _react2.default.Children.only(children);
        return _react2.default.createElement(
          'li',
          null,
          _react2.default.createElement(
            Element,
            _extends({ className: (0, _classnames2.default)(className, { 'is-active': active }) }, props),
            child.props.title
          ),
          _react2.default.cloneElement(child, { title: null })
        );
      }

      return _react2.default.createElement(
        'li',
        null,
        children
      );
    }
  }]);

  return MenuListItem;
}(_react.PureComponent);

MenuListItem.displayName = 'Menu.List.Item';
MenuListItem.propTypes = {
  className: _propTypes2.default.string,
  children: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.element]),
  active: _propTypes2.default.bool,
  renderAs: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.func])
};
MenuListItem.defaultProps = {
  className: '',
  children: null,
  active: false,
  renderAs: 'a'
};
exports.default = MenuListItem;
