// react-bulma-components/lib/components/breadcrumb/breadcrumb.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Breadcrumb = function (_PureComponent) {
  _inherits(Breadcrumb, _PureComponent);

  function Breadcrumb() {
    _classCallCheck(this, Breadcrumb);

    return _possibleConstructorReturn(this, (Breadcrumb.__proto__ || Object.getPrototypeOf(Breadcrumb)).apply(this, arguments));
  }

  _createClass(Breadcrumb, [{
    key: 'render',
    value: function render() {
      var _classnames;

      var _props = this.props,
          className = _props.className,
          items = _props.items,
          renderAs = _props.renderAs,
          hrefAttr = _props.hrefAttr,
          separator = _props.separator,
          size = _props.size,
          align = _props.align,
          props = _objectWithoutProperties(_props, ['className', 'items', 'renderAs', 'hrefAttr', 'separator', 'size', 'align']);

      if (renderAs !== 'a' && !hrefAttr) {
        // eslint-disable-next-line no-console
        console.warn('if renderAs is different the anchor (a), hrefAttr is required. Check Breadcrumb props');
      }
      var Element = renderAs;
      return _react2.default.createElement(
        'nav',
        _extends({}, props, {
          className: (0, _classnames3.default)('breadcrumb', className, (_classnames = {}, _defineProperty(_classnames, 'has-' + separator + '-separator', separator), _defineProperty(_classnames, 'is-' + size, size), _defineProperty(_classnames, 'is-' + align, align), _classnames))
        }),
        _react2.default.createElement(
          'ul',
          null,
          items.map(function (item) {
            var p = _defineProperty({}, renderAs === 'a' ? 'href' : hrefAttr, item.url);
            return _react2.default.createElement(
              'li',
              {
                key: item.url,
                className: (0, _classnames3.default)({
                  'is-active': item.active
                })
              },
              _react2.default.createElement(
                Element,
                p,
                item.name
              )
            );
          })
        )
      );
    }
  }]);

  return Breadcrumb;
}(_react.PureComponent);

Breadcrumb.propTypes = {
  className: _propTypes2.default.string,
  style: _propTypes2.default.shape({}),
  separator: _propTypes2.default.oneOf(['arrow', 'bullet', 'dot', 'succeeds']),
  size: _propTypes2.default.oneOf(['small', 'medium', 'large']),
  align: _propTypes2.default.oneOf(['right', 'center']),
  items: _propTypes2.default.arrayOf(_propTypes2.default.shape({
    url: _propTypes2.default.string.isRequired,
    active: _propTypes2.default.bool,
    name: _propTypes2.default.node
  })),
  renderAs: _propTypes2.default.oneOfType([_propTypes2.default.oneOf(['a']), _propTypes2.default.func]),
  hrefAttr: _propTypes2.default.string
};
Breadcrumb.defaultProps = {
  items: [],
  hrefAttr: null,
  separator: null,
  renderAs: 'a',
  className: '',
  style: {},
  size: null,
  align: null
};
exports.default = Breadcrumb;
