// react-bulma-components/lib/components/dropdown/dropdown.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

var _constants = require('../../constants');

var _constants2 = _interopRequireDefault(_constants);

var _item = require('./components/item');

var _item2 = _interopRequireDefault(_item);

var _divider = require('./components/divider');

var _divider2 = _interopRequireDefault(_divider);

var _button = require('../button');

var _button2 = _interopRequireDefault(_button);

var _icon = require('../icon');

var _icon2 = _interopRequireDefault(_icon);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var colors = [null].concat(Object.keys(_constants2.default.COLORS).map(function (key) {
  return _constants2.default.COLORS[key];
}));

var Dropdown = function (_PureComponent) {
  _inherits(Dropdown, _PureComponent);

  function Dropdown(props) {
    _classCallCheck(this, Dropdown);

    var _this = _possibleConstructorReturn(this, (Dropdown.__proto__ || Object.getPrototypeOf(Dropdown)).call(this, props));

    _initialiseProps.call(_this);

    _this.checkProps(props);
    return _this;
  }

  _createClass(Dropdown, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      document.addEventListener('click', this.close);
    }
  }, {
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      this.checkProps(_extends({}, this.props, nextProps));
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      document.removeEventListener('click', this.close);
    }
  }, {
    key: 'render',
    value: function render() {
      var _this2 = this,
          _classnames;

      var _props = this.props,
          className = _props.className,
          children = _props.children,
          value = _props.value,
          color = _props.color,
          align = _props.align,
          hoverable = _props.hoverable,
          onChange = _props.onChange,
          props = _objectWithoutProperties(_props, ['className', 'children', 'value', 'color', 'align', 'hoverable', 'onChange']);

      var current = null;

      var childrenArray = _react2.default.Children.map(children, function (child, i) {
        if (i === 0 || child.props.value === value) {
          current = child.props.children;
        }
        return _react2.default.cloneElement(child, child.type === _item2.default ? {
          active: child.props.value === value,
          onClick: _this2.select(child.props.value)
        } : {});
      });

      return _react2.default.createElement(
        'div',
        _extends({}, props, {
          ref: function ref(node) {
            _this2.htmlElement = node;
          },
          className: (0, _classnames3.default)('dropdown', className, (_classnames = {
            'is-active': this.state.open
          }, _defineProperty(_classnames, 'is-' + align, align), _defineProperty(_classnames, 'is-hoverable', hoverable), _classnames))
        }),
        _react2.default.createElement(
          'div',
          { className: 'dropdown-trigger', role: 'presentation', onClick: this.toggle },
          _react2.default.createElement(
            _button2.default,
            { color: color },
            _react2.default.createElement(
              'span',
              null,
              current
            ),
            _react2.default.createElement(_icon2.default, { icon: 'angle-down', size: 'small' })
          )
        ),
        _react2.default.createElement(
          'div',
          { className: 'dropdown-menu', id: 'dropdown-menu', role: 'menu' },
          _react2.default.createElement(
            'div',
            { className: 'dropdown-content' },
            childrenArray
          )
        )
      );
    }
  }]);

  return Dropdown;
}(_react.PureComponent);

Dropdown.propTypes = {
  className: _propTypes2.default.string,
  style: _propTypes2.default.shape({}),
  children: _propTypes2.default.node,
  value: _propTypes2.default.any,
  onChange: _propTypes2.default.func,
  color: _propTypes2.default.oneOf(colors),
  align: _propTypes2.default.oneOf(['right']),
  hoverable: _propTypes2.default.bool
};
Dropdown.defaultProps = {
  className: '',
  style: {},
  value: null,
  children: [],
  onChange: null,
  color: null,
  align: null,
  hoverable: false
};
Dropdown.Item = _item2.default;
Dropdown.Divider = _divider2.default;

var _initialiseProps = function _initialiseProps() {
  var _this3 = this;

  this.state = {
    open: false
  };

  this.checkProps = function (props) {
    if (props.value && !props.onChange) {
      // eslint-disable-next-line no-console
      console.error('Warning: This is a controlled component without onChange listener, please check the props of the Dropdown component');
    }
  };

  this.close = function (evt) {
    // IDK yet how to test using the ref in enzime
    if (_this3.props.hoverable || evt && evt.path.find(function (node) {
      return node === _this3.htmlElement;
    })) {
      return;
    }
    _this3.setState({ open: false });
  };

  this.toggle = function (evt) {
    if (_this3.props.hoverable) {
      return;
    }
    if (evt) {
      evt.preventDefault();
    }
    _this3.setState({ open: !_this3.state.open });
  };

  this.select = function (value) {
    return function () {
      if (_this3.props.onChange) {
        _this3.props.onChange(value);
      }
      _this3.close();
    };
  };
};

exports.default = Dropdown;
